gateway/fedimint-lightning/src/ldk.rs

```diff
@@ -513,27 +560,16 @@ impl ILnRpcClient for GatewayLdkClient {
                 failure_reason: e.to_string(),
             })?;
 
-        // The channel isn't always visible immediately, so we need to poll for it.
-        for _ in 0..10 {
-            let funding_txid_or = self
-                .node
-                .list_channels()
-                .iter()
-                .find(|channel| channel.user_channel_id == user_channel_id)
-                .and_then(|channel| channel.funding_txo)
-                .map(|funding_txo| funding_txo.txid);
-
-            if let Some(funding_txid) = funding_txid_or {
-                return Ok(OpenChannelResponse {
-                    funding_txid: funding_txid.to_string(),
-                });
-            }
-
-            fedimint_core::runtime::sleep(Duration::from_millis(100)).await;
-        }
+        self.add_pending_channel(user_channel_id, tx).await;
```

Comment:
I think we need to add the oneshot channel before opening the LN channel, otherwise there could be a race condition where the pending channel event is fired before we are ready to receive it.

Reply:
Good catch. I changed the code to acquire the `RwLock` before opening the channel, so that the event loop will block when acquiring the lock if the sender has not been added to the map yet.